### PR TITLE
[WIP] THE Frontend Core Skill

### PR DESCRIPTION
**Pull request by SebAshton**

### CODE REVIEW

File: core-skills/frontend-web-development/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Frontend web developement [HTML/CSS]
+
+## Aim
+This course is intended to guide learning to a level where you can write and contribute semanitic, and maintainable HTML and CSS to a project.
```
Comment (craigjbass) [inline core-skills/frontend-web-development/README.md]: Semantic

File: core-skills/frontend-web-development/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The course guides you through a typical "first page" build of any green field project. This encompasses selecting/creating a responsive CSS grid, building core shared components for your new site, and finally delivering an accessible, SEO optimised, lightning fast loading product detail page.
+
+## Assessment Style
+This core skill assessment is split into two parts. Firstly, you'll take a design from PSD to HTML/CSS during Learn Tech. This is assessed over three levels; Rhino, Narwhal, and Unicorn. Secondly, before completing each level as HTML/CSS is quite subjective you'll be asked to talk about the decisions you made whilst building out the deliverables for each level.
```
Comment (craigjbass) [inline core-skills/frontend-web-development/README.md]: Not sure about PSD - we should probs only require from PNG or PDF.

File: core-skills/frontend-web-development/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+- Pixel Perfect HTML/CSS (Your preferred browser)
+- Responsive Design
+- Cross Browser testing (Chrome/Firefox/Edge/Safari)
+- Semantic HTML (i.e. no [Tag Soup](https://en.wikipedia.org/wiki/Tag_soup) )
+- Logical CSS naming
+
+It is up to the assessor to determine whether or not the candidate has adequetly achieved the above in addition to the deliverables. 
+
+As this isn't an assessment about scaffolding an application, and gathering resources we have built the (base application)[https://github.com/madetech/core-skill-frontend]. At this time this repo is private to Made Tech members, we're working to open source this too.
```
Comment (craigjbass) [inline core-skills/frontend-web-development/README.md]: I noticed this private repo scaffolds font tags and other icons. I assume these are proprietary? Also does this assessment cover being able to write that code?

File: core-skills/frontend-web-development/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Core Concepts
+Each of the sections below contain reading material and links to related resources and tools to help you on your way. Some of them wont be assessed but are good things to know about.
+
+- [Grids](#)
```
Comment (craigjbass) [inline core-skills/frontend-web-development/README.md]: Do we have content for these areas yet or empty?

File: core-skills/frontend-web-development/README.md (hunk 2 above)
Comment (SebAshton) [reply core-skills/frontend-web-development/README.md]: On a delivery, you'd typically get a PSD or a Sketch file. I feel that frontend engineers should be comfortable working with the format

File: core-skills/frontend-web-development/README.md (hunk 4 above)
Comment (SebAshton) [reply core-skills/frontend-web-development/README.md]: We have them in a doc, just need to format

File: core-skills/frontend-web-development/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+- Accessible
+- Page load optimisations
+
+While presenting their pages, candidates should explain the following:
+
+- Did you have to change any of your previous HTML to follow accessibility guidelines?
+- What techniques did you use to speed up the page load?
+- How did you keep your form accessible whilst applying custom styling?
+- If you had to stop flash of unstyled content (FOUC) how would you have gone about it?
+- If a client requested the page structure to remain consistant while the page loaded how would you do that?
```
Comment (zuren) [inline core-skills/frontend-web-development/README.md]: Don’t fully understand this question. Does it imply it won’t? What do you mean by page structure?

File: core-skills/frontend-web-development/resources/grids.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Grids
+
+Long gone are the days of table based layouts. Engineers use CSS to achieve basic to the most complex of page layouts.
+
+Understanding of the CSS box model (margin, border, padding etc) should be essential too. Need to get that in here?
```
Comment (zuren) [inline core-skills/frontend-web-development/resources/grids.md]: “Need to get that in here?”? WIP comment?

File: core-skills/frontend-web-development/README.md (hunk 2 above)
Comment (craigjbass) [reply core-skills/frontend-web-development/README.md]: True - I’d argue that good web designers wouldn’t use PSDs but that’s a separate problem to what this PR is attempting to solve

File: core-skills/frontend-web-development/README.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+- Maintainable well architected CSS
+
+**Javascript is not required to complete any level.**
+
+### Rhino
+
+To obtain rhino a candidate has to present the following to the examiner:
+
+- A grid
+- A sticky header (remembering we're not testing JS)
```
[the remark this answers is not in the document]
Comment (craigjbass) [reply core-skills/frontend-web-development/README.md]: I think the use of testing is confusing - perhaps “assessing”.

File: core-skills/frontend-web-development/README.md (hunk 5 above)
Comment (craigjbass) [reply core-skills/frontend-web-development/README.md]: Perhaps there should be something in here about demonstrating simulating a slow connection ? And the impact on UX?

Comment (craigjbass) [reply core-skills/frontend-web-development/README.md]: Btw just the worst on news sites that break scroll position cos of adverts that pop out height

File: core-skills/frontend-web-development/README.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
+# Frontend web development [HTML/CSS]
+
+## Aim
+This course is intended to guide learning to a level where you can write and contribute semantic, and maintainable HTML and CSS to a project.
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/README.md]: delivery > project

File: core-skills/frontend-web-development/README.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+The course guides you through a typical "first page" build of any green field project. This encompasses selecting/creating a responsive CSS grid, building core shared components for your new site, and finally delivering an accessible, SEO optimised, lightning fast loading product detail page.
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/README.md]: application > site

File: core-skills/frontend-web-development/README.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+- Creation of any additional assets
+
+While presenting their pages, candidates should explain the following:
+
+- What image format did you pick for Logo
+- What are the different values of `position` and what do they do.
+- What are the different values of `display` and what do they do.
+- What tools if any did you use to make the icons?
+- Their initial CSS structure, and how they plan to keep it maintainable?
+- Their choice to use (or not use) an off the shelf grid?
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/README.md]: How are things like this "marked"?

File: core-skills/frontend-web-development/README.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+- Main product image, and thumbnails
+- SEO
+
+While presenting their pages, candidates should explain the following:
+
+- Did you have to DRY up any CSS as you found common patterns?
+- What additional SEO features have you added? (HINT: Microformats)?
+- How would open graph meta tags help discoverability?
+- Which image format have you chosen to use? How would you optimize them if you had to?
+- The client wants a new promotion on the page and the design shows white text on a very light background. What would you recommend in this situation? If anything?
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/README.md]: Customer > Client

File: core-skills/frontend-web-development/resources/grids.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+Without using a grid system for creating a layout, there are:
+
+Floats (including clear for vertical layout considerations)
+Positioning (including position relative and absolute)
+Flexbox (including flex-direction for vertical layout considerations)
+
+Engineer can describe each of these and think of both positives and negatives for using each.
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/resources/grids.md]: think of or describe?

File: core-skills/frontend-web-development/resources/grids.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+Do we want to list some of the pros/cons that we think there might be? Not all, but some might be subjective.
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/resources/grids.md]: WIP?

File: core-skills/frontend-web-development/resources/grids.md [hunk 14]
```diff
@@ -0,0 +1,32 @@
+# Grids
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/resources/grids.md]: Generally unsure what this guidance is trying to solve

File: core-skills/frontend-web-development/README.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+In addition to the deliverables, at each level you are expected to demonstrate:
+
+- Pixel Perfect HTML/CSS (Your preferred browser)
+- Responsive Design
+- Cross Browser testing (Chrome/Firefox/Edge/Safari)
+- Semantic HTML (i.e. no [Tag Soup](https://en.wikipedia.org/wiki/Tag_soup) )
+- Logical CSS naming
+
+It is up to the assessor to determine whether or not the candidate has adequetly achieved the above in addition to the deliverables.
```
Comment (chrisblackburn) [inline core-skills/frontend-web-development/README.md]: How can this not be subjective?

**Comment by chrisblackburn** [comment]
Is this lacking a marking guide as demonstrated by the TDD core skill? Unclear for any of the criteria in here what would constitute a pass.

This example feels good (https://github.com/madetech/learn/blob/master/core-skills/tdd/well-designed-tests.md) in that it's clear what is expected to pass, but avoids being overly prescriptive.

**Comment by craigjbass** [comment]
I think with reference to this RFC https://github.com/madetech/rfcs/blob/master/rfc-010-core-skill-assessments.md the marking guide can certainly be tightened up.

**Comment by craigjbass** [comment]
@SebAshton - I think we should break some of the issues that would take a bit more time related to Narwhal/Unicorn out into separate discussions - https://github.com/madetech/learn/issues. 

As mentioned, we should focus on Rhino being ready for prime time.

File: core-skills/frontend-web-development/README.md (hunk 3 above)
Comment (SebAshton) [reply core-skills/frontend-web-development/README.md]: Candidates should be aware of how to make an iconfont if they need it, however writing an `@font-face` declaration is pretty trivial and easily google'able so I feel we can give them that

File: core-skills/frontend-web-development/README.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+- Semantic HTML
+- Maintainable well architected CSS
+
+**Javascript is not required to complete any level.**
+
+### Rhino
+
+To obtain rhino a candidate has to present the following to the examiner:
+
+- A grid
```
[the remark this answers is not in the document]
Comment (SebAshton) [reply core-skills/frontend-web-development/README.md]: No not at all, we will amend to explicitly state this.

File: core-skills/frontend-web-development/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (SebAshton) [reply core-skills/frontend-web-development/README.md]: @zuren  We feel a candidate should be able to achieve this with CSS only, again we'll explicitly state _no JS_ required.

File: core-skills/frontend-web-development/README.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+- A sticky header (remembering we're not testing JS)
+- Footer
+- Creation of any additional assets
+
+While presenting their pages, candidates should explain the following:
+
+- What image format did you pick for Logo
+- What are the different values of `position` and what do they do.
+- What are the different values of `display` and what do they do.
```
Comment (craigjbass) [inline core-skills/frontend-web-development/README.md]: There's a lot of values - 23 - I'm not sure anyone can remember this many.

**Comment by craigjbass** [comment]
Can we get this merged this week?

**Pull request merged**